src/events/etcd.py

```diff
@@ -189,8 +197,11 @@ def _on_start(self, event: ops.StartEvent) -> None:  # noqa: C901
 
     def _on_config_changed(self, event: ops.ConfigChangedEvent) -> None:
         """Handle config_changed event."""
-        if self.charm.state.cluster.is_restore_in_progress:
-            logger.warning("Cannot update config while database restore is in progress.")
+        if (
+            self.charm.state.cluster.is_restore_in_progress
+            or self.charm.state.cluster.rebuild_cluster_in_progress
+        ):
+            logger.warning("Cannot update config while another operation is in progress.")
```

Comment:
I would perhaps be more specific here, something along the lines of:
```suggestion
            logger.warning("Cannot update config while a restore or cluster-rebuild operation is in progress.")
```

Reply:
updated